src/opm/output/eclipse/Summary.cpp

```diff
@@ -1380,54 +1380,29 @@ void eval_udq(const Opm::Schedule& schedule, std::size_t sim_step, Opm::SummaryS
     const UDQConfig& udq = schedule.getUDQConfig(sim_step);
     const auto& func_table = udq.function_table();
     UDQContext context(func_table, st);
-    {
-        const std::vector<std::string> wells = st.wells();
-
-        for (const auto& assign : udq.assignments(UDQVarType::WELL_VAR)) {
-            auto ws = assign.eval(wells);
-            for (const auto& well : wells) {
-                const auto& udq_value = ws[well];
-                if (udq_value)
-                    st.update_well_var(well, ws.name(), udq_value.value());
-            }
-        }
+    for (const auto& assign : udq.assignments(UDQVarType::WELL_VAR)) {
+        auto ws = assign.eval(st.wells());
```

Comment:
This calls [`SummaryState::wells()`](https://github.com/OPM/opm-common/blob/67f935a502019c1c8a9ab38a6f94968aa8aa6e98/src/opm/parser/eclipse/EclipseState/Schedule/SummaryState.cpp#L248-L250) in a loop which generates a new `vector<string>`&mdash;albeit with the same values&mdash;for each call.  Does that have any noticeable performance impact?